### DISCUSSION TITLE
Why Closed Source?

### DISCUSSION THREAD

**Discussion by blaze-developer** [vinegarhq/sober#199] [General]
The fact that this project is closed-source is a major red flag. And leaves room for malware. Making this software seem untrustable for many users. I am sure many people can think of no reason to keep such a project closed source, perhaps other than simply not wanting to deal with pull requests. If I may inquire, why is this project closed source?

I come with no offense, no accusations, and am simply stating my opinion on this matter, hopefully to get a response.

Thank you,

**Comment by Amgsdfyhadfgybi** [comment]
its closed source probably so they can be "undetected" by roblox

**Comment by Amgsdfyhadfgybi** [comment]
and its also not malware, i use it

**Reply by sergiointel** [reply]
That's no valuable response, it can very well be silent and persistent.

**Comment by adverbialsatz** [comment]
You can talk to the devs in the discord server. They are pretty transparent and I run it as well. It is not malware

**Reply by dragoonmaster999** [reply]
where's the discord server?

**Comment by VoidusX** [comment]
> The fact that this project is closed-source is a major red flag. And leaves room for malware. Making this software seem untrustable for many users.

Hello, I am one of the roblox developers that use Sober for daily use and I want to clarify some things up regarding the project being closed source. It is impossible to tamper with the current implementations by design, including the use of flatpak's sandboxing. The chances of getting malware is near impossible unless you are either getting the project from a untrusted source or you manually installed a apk from a untrusted site.

> I am sure many people can think of no reason to keep such a project closed source, perhaps other than simply not wanting to deal with pull requests. If I may inquire, why is this project closed source?

Sober was decided to be closed source in order to respect roblox's wishes, and to prevent a certain community from taking advantage of the software for malicious purposes. While this is something that will not please anyone, this was a decision made within development of the project at hand.

I hope this explains well behind the decision to make this project closed source. This has been asked before in other places, and these answers from the maintainers are no different from the others.

**Reply by blaze-developer** [reply]
Escaping the flatpak container is simply trivial for software. And I am not concerned about getting the software from an *untrusted* souce, I am concerned about the software itself.

**Reply by BrycensRanch** [reply]
> Escaping the flatpak container is simply trivial for software. And I am not concerned about getting the software from an _untrusted_ souce, I am concerned about the software itself.

If it is as trivial as you say, publish a GitHub repository with reproducible steps. Cheers

**Reply by blaze-developer** [reply]
I am not here for an argument. Anyone can have the right to not trust a piece of software. No matter what security they boast.

**Reply by BrycensRanch** [reply]
> I am not here for an argument. Anyone can have the right to not trust a piece of software. No matter what security they boast.

Then don't use it - I trust the sober devs and I hope it will be open-source one day. But it's important to know you are not entitled to the source code.

**Comment by Naymmmm** [comment]
Devs will skid off the source and create exploits, forcing Roblox to implement Hyperion onto the Android versions resulting in no more Sober.

**Reply by Amgsdfyhadfgybi** [reply]
the devs hes talking about are script kiddies that modify the game to add a way to use modified apks, hence resulting in people using modified apks with hacks in them

**Reply by blaze-developer** [reply]
How would that even be possible? The way this project is set up explicitly doesn't change the Roblox client, only provides a way to run it. If they actually could make changes to the Roblox client, they would actually fix bugs rather than wait for Roblox to fix bugs for android.

**Reply by existentials** [reply]
It's possible because by definition, Roblox runs in Sober's address space, using Sober's implementation for the system runtime, all for interoperability reasons. It works differently to something like Bloxstrap, where it's very obviously just a program that unpacks Roblox and runs it. Given this tight integration, and given our technical due diligence, we figured the project would last longer as a closed-source blob. (I think we were right)

**Reply by blaze-developer** [reply]
> It's possible because by definition, Roblox runs in Sober's address space, using Sober's implementation for the system runtime, all for interoperability reasons. It works differently to something like Bloxstrap, where it's very obviously just a program that unpacks Roblox and runs it. Given this tight integration, and given our technical due diligence, we figured the project would last longer as a closed-source blob. (I think we were right)

You could say that for everything. It's possible to cheat in x because it runs on someones PC that they can install programs on! It's in their operating systems address space! Oh no! Do you see what I'm trying to say?

No offense to you guys and the work you probably do, I'm just trying to acknowledge the fact that a *community* made software, specifically for Linux support being closed source to comply with what a company might do is kind of ironic.

I appreciate your commitment to the community, that's just what I would like to say.

**Comment by blaze-developer** [comment]


**Reply by BrycensRanch** [reply]
Weird hill to die on but ok. Believe it or not, the vinegarhq developers do not get paid per user.
